### ISSUE TITLE
Tenants with non-standard 365 REST URLs return the default 365 REST URL in Outlook on iOS

### ISSUE THREAD

**Issue by GrumpyAL** [OfficeDev/office-js#1888]
## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: Outlook App on iOS
* Host [Excel, Word, PowerPoint, etc.]: Outlook
* Office version number: n/a (any tested so far.  Apologies, I don't have one available).
* Operating System: iOS
* Browser (if using Office on the web): n/a
 
## Expected behavior
If the 365 tenant has a custom or vanity mail URL in addition to the defaults (outlook.office.com or outlook.office365.com) then this custom URL is returned by Office.context.mailbox.restUrl and matches the audience URL inside of the REST token obtained by Office.context.mailbox.getCallbackTokenAsync({isRest:true}, callback) when using Outlook on iOS.

## Current behavior

If the 365 tenant has a custom or vanity mail URL in addition to the defaults (outlook.office.com or outlook.office365.com) the **default** URL is returned by Office.context.mailbox.restUrl and does not match the audience URL inside of the token (the token has the custom URL) obtained by Office.context.mailbox.getCallbackTokenAsync({isRest:true}, callback) when using Outlook on iOS.

## Steps to reproduce
I do not have a tenant with a custom/vanity URL so I cannot reproduce this for myself.  However, the steps are as follows:

1. On a tenant with a custom mail URL, open Outlook on iOS
2. Run Office.context.mailbox.RestUrl to get the rest URL
3. Run Office.context.mailbox.getCallbackTokenAsync(isRest:true, callback) to get the token.
4. Inspect the token in JWT.IO
5. Compare the aud claim in the token the the REST URL returned by the framework
6. Attempt to use the token with rest URL returned by the framework and it will be denied.

## Link to live example(s)
none

# Provide additional details

## Context
This appears to only occur on iOS.  Outlook on Android and non-Mobile clients works as expected, the URLs match.

**Comment by exextoc** [comment]
Hello, thanks for reporting this issue. This is however a known issue and we are working to release a fix soon.

But as a workaround, please use the Audience Field inside of the JWT Tokens received from getCallbackTokenAsync. (Callback tokens are JWT tokens and can be decoded using libraries such as these: https://github.com/auth0/jwt-decode). You can use this Audience field to make further API calls.


```
// https://www.npmjs.com/package/jwt-decode
import jwt_decode from "jwt-decode";


// Use this as a fallback for 401


Office.context.mailbox.getCallbackTokenAsync({ isRest: true }, function (result) {
var ewsId = Office.context.mailbox.item.itemId;
var token = result.value;
var decoded = jwt_decode(token);
var restId = Office.context.mailbox.convertToRestId(ewsId, Office.MailboxEnums.RestVersion.v2_0);
var audienceURL = decoded.aud
//Create the Endpoint URL which you want to call
var getMessageUrl = audienceURL + '/v2.0/me/messages/' + restId;
var xhr = new XMLHttpRequest();
xhr.open('GET', getMessageUrl);
xhr.setRequestHeader("Authorization", "Bearer " + token);
xhr.onload = function (e) {
console.log(this.response);
}
xhr.send();
});
```

**Comment by GrumpyAL** [comment]
Hello,

I have heard from someone else that a fix for this has started to roll out.  Is that true? (I, unfortunately, have no way to test this). You previously stated: 

> This is however a known issue and we are working to release a fix soon.

However, I wasn't sure if/when that may make it out to production Outlook for iOS clients (especially given that the REST API is deprecated).  Any update appreciated!  Thanks!

**Comment by exextoc** [comment]
Hey @GrumpyAL ,

Yes, the fix has been released, and will be available in iOS app 4.2142.0 onwards.

**Comment by GrumpyAL** [comment]
@exextoc I am starting to see this issue crop up again.  At least for Android, have not been able to confirm for iOS yet.
